packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs

```diff
@@ -29,9 +29,22 @@ public async Task ExecuteAsync()
             var generatedTestOutputPath = CodeModelPlugin.Instance.Configuration.TestGeneratedDirectory;
 
             GeneratedCodeWorkspace workspace = await GeneratedCodeWorkspace.Create();
+            var output = CodeModelPlugin.Instance.OutputLibrary;
+
+            // The generated attributes need to be added into the workspace before loading the custom code. Otherwise,
+            // Roslyn doesn't load the attributes completely and we are unable to get the attribute arguments.
+
+            List<Task> generateAttributeTasks = new();
+            foreach (var attributeProvider in output.CustomCodeAttributeProviders)
+            {
+                var writer = CodeModelPlugin.Instance.GetWriter(attributeProvider);
+                generateAttributeTasks.Add(workspace.AddGeneratedFile(writer.Write()));
+            }
+
+            await Task.WhenAll(generateAttributeTasks);
+
             CodeModelPlugin.Instance.SourceInputModel = new SourceInputModel(await workspace.GetCompilationAsync());
 
-            var output = CodeModelPlugin.Instance.OutputLibrary;
```

Comment:
Lets move this back down here by directly accessing the definition classes to create the custom attributes into the workspace.  This will avoid any order of operations issues.